packages/sentry/sentry.js

```diff
@@ -1,68 +1,83 @@
-    }).install();
+    });
+    Sentry.configureScope(scope => {
+      scope.setTag('role', 'frontend');
+    });
+  }
 };
 
 export const updateUser = user => {
   if (user && window.app.env === 'production') {
     // to avoid sending personal data to sentry we anonymize the email address
     // by only sending the domain part or the email
     const emailTld = user.email.split('@')[1];
-    Raven.setUserContext({
-      email: `xxx@${emailTld}`,
-      id: user.id,
+    Sentry.configureScope(scope => {
+      scope.setUser({
+        email: `xxx@${emailTld}`,
+        id: user.id,
+      });
     });
   }
 };
 
 export const stopTrackingUser = () => {
-  if (window.app.env === 'production') Raven.setUserContext();
+  if (window.app.env === 'production')
+    Sentry.configureScope(scope => {
+      scope.clear();
+    });
 };
 
-// NOTE: only exported for tests => use reportError below for your code
-export const createSentryErrorReporter = (
-  environment,
-  debugLogger = console,
-  productionLogger = Raven
-) => (error, extraInfo) => {
-  if (error instanceof Error === false && environment === 'development') {
-    debugLogger.warn(
-      '[SENTRY]: You called "sentry.reportError" with an argument that is not an instance of "Error". ' +
+export const reportErrorToSentry = (error, extraInfo, getIsEnabled) => {
+  const environment = window.app.env || process.env.NODE_ENV;
+  const isEnabled = getIsEnabled
+    ? getIsEnabled(environment)
+    : environment === 'production';
+
+  if (error instanceof Error === false && !isEnabled) {
+    console.warn(
+      '[SENTRY]: You called "reportErrorToSentry" with an argument that is not an instance of "Error". ' +
         '"Error" should be preferred so that a stack-trace can be made available in sentry. ' +
         'See: https://docs.sentry.io/clients/javascript/usage/#try-catch'
     );
   }
 
-  if (environment === 'production') {
+  if (isEnabled) {
     // logs error to sentry
-    if (error instanceof Error) {
-      productionLogger.captureException(error, extraInfo);
-    } else {
-      productionLogger.captureMessage(error, extraInfo);
-    }
-
-    // Generate a unique ID referring to the last generated Sentry error
-    const errorId = productionLogger.lastEventId();
-
-    // The error stack should be available in Sentry, so there is no
-    // need to print it in the console as well.
-    // We just notify that an error occurred and provide the error ID.
-    debugLogger.error(`[SENTRY]: An error occured (ID: ${errorId}).`);
+    Sentry.withScope(scope => {
+      if (extraInfo && extraInfo.extra) {
+        if (typeof extraInfo.extra === 'object') {
+          // See https://docs.sentry.io/platforms/javascript/react/
+          Object.keys(extraInfo.extra).forEach(key => {
+            scope.setExtra(key, extraInfo.extra[key]);
+          });
+        } else {
+          scope.setExtra('extra', extraInfo.extra);
```

Comment:
In case the `extra` info is not an object, we pass it as a value under the key `extra`